test/e2e/accounts/create-snap-account.spec.ts

```diff
@@ -183,7 +188,112 @@ describe('Create Snap Account', function (this: Suite) {
     );
   });
 
-  it('create Snap account confirmation cancelation results in error in Snap', async function () {
+  it('create Snap account confirmation flow ends in approval success with custom name input', async function () {
+    await withFixtures(
+      {
+        fixtures: new FixtureBuilder().build(),
+        ganacheOptions: defaultGanacheOptions,
+        title: this.test?.fullTitle(),
+      },
+      async ({ driver }: { driver: Driver }) => {
+        await unlockWallet(driver);
+
+        // navigate to test Snaps page and connect
+        await driver.openNewPage(TEST_SNAPS_SIMPLE_KEYRING_WEBSITE_URL);
+        await driver.clickElement('#connectButton');
+
+        // switch to metamask extension and click connect to start installing the snap
+        await switchToNotificationWindow(driver);
+        await driver.clickElement({
+          text: 'Connect',
+          tag: 'button',
+        });
+
+        // scroll to the bottom of the page
+        await driver.waitForSelector({ text: 'Confirm' });
+        await driver.clickElementSafe('[data-testid="snap-install-scroll"]');
+
+        // click the install button to install the snap
+        await driver.waitForSelector({ text: 'Confirm' });
+        await driver.clickElement({
+          text: 'Confirm',
+          tag: 'button',
+        });
+        await driver.waitForSelector({ text: 'OK' });
+        await driver.clickElement({
+          text: 'OK',
+          tag: 'button',
+        });
+
+        // move back to the Snap window to test the create account flow
+        await driver.switchToWindowWithTitle(
+          WINDOW_TITLES.SnapSimpleKeyringDapp,
+        );
+
+        // check the dapp connection status
+        await driver.waitForSelector({
+          css: '#snapConnected',
+          text: 'Connected',
+        });
+
+        // create new account on dapp
+        await driver.clickElement({
+          text: 'Create account',
+          tag: 'div',
+        });
+
+        await driver.clickElement({
+          text: 'Create Account',
+          tag: 'button',
+        });
+
+        await switchToNotificationWindow(driver);
+
+        await driver.clickElement('[data-testid="confirmation-submit-button"]');
```

Comment:
What about creating a function for this like startCreateSnapAccountFlow (not sure about my naming, but the function should make it clear that the "full flow" requires also the account name to be submitted, hence this function is not "sufficient").

Since we use the same exact flow for the other e2e test cancelling naming Snap account results in account not created having a function makes sense to me.

Reply:
I agree this would be very helpful 😄 I added the helper function here [1d1b3ae](https://github.com/MetaMask/metamask-extension/pull/25191/commits/1d1b3ae517b13e85b82cb04be035a5d57f46f496)